website/static/js/components/autocomplete.js

```diff
@@ -0,0 +1,171 @@
+        throw new MethodNotDefined('suggestionTemplate');
+    },
+    /** Called when the form is submitted */
+    onSubmit: function() {
+        throw new MethodNotDefined('onSubmit');
+    },
+    /** Get the data at the provided URL, return a promise */
+    fetchData: function () {
+        return $.get(this.dataUrl).fail(
+            function() {
+                console.log('failed to retrieve data');
+            });
+    },
+    /** Initialize the Typeahead widget for the component */
+    initTypeahead: function () {
+        this.inputElement.typeahead({
+            hint: false,
+            highlight: true,
+            minLength: 0
+        }, {
+            // name: 'projectSearch' + nodeType + namespace,
+            displayKey: function(data) {
+                return data.value.node.title;
+            },
+            templates: {
+                suggestion: this.suggestionTemplate
+            },
+            source: this.substringMatcher(this.items())
+        });
+
+        this.inputElement.bind('typeahead:selected', function(event, item) {
+            // Call the parent viewModel's onSelected
+            this.onSelected(item.value);
+        }.bind(this));
+    },
+    /** Clear the selection */
+    onClear: function () {
+        this.inputElement.typeahead('val', '');
+        this.itemSelected(null);
+    },
+    /** Set the item as selected */
+    onSelected: function (item) {
+        this.itemSelected(item);
+    }
+});
+
+
+var draftRegistrationsSearchViewModel = function (params) {
+    baseSearchViewModel.apply(this, arguments);
+};
+$.extend(draftRegistrationsSearchViewModel.prototype, baseSearchViewModel.prototype, {
+    processData: function (data) {
+        this.items(data.draftRegistrations);
+    },
+    onSubmit: function () {
+        // Redirect the user to continue the draft registration
+        window.location = this.itemSelected().url;
+    },
+    suggestionTemplate: function(item) {
+        var dateUpdated = new $osf.FormattableDate(item.value.dateUpdated);
+        var dateCreated = new $osf.FormattableDate(item.value.dateCreated);
+        return '<p>' + item.value.node.title + '</p>' +
+            '<p><small class="m-l-md text-muted">' + 'Initiated by: ' + item.value.initiator.name + '</small></p>' +
+            '<p><small class="m-l-md text-muted">' + 'Initiated: ' + dateCreated.local + '</small></p>' +
+            '<p><small class="m-l-md text-muted">' + 'Last updated: ' + dateUpdated.local + '</small></p>';
+    },
+    substringMatcher: function(strs) {
+        return function findMatches(q, cb) {
+            var matches = [];
+            var substrRegex = new RegExp(q, 'i');
+            var count = 0;
+
+            $.each(strs, function(i, str) {
+                if (substrRegex.test(str.node.title)) {
+                    count += 1;
+                    matches.push({ value: str });
+
+                    //alex's hack to limit number of results
```

Comment:
Is this hack necessary? Are other options (such as scrollbar technology) available?